### ISSUE TITLE
"cargo package" crash: "slice index starts at 102 but ends at 101"

### ISSUE THREAD

**Issue by jsgf** [rust-lang/cargo#2749]
`cargo package` fails with `thread '<main>' panicked at 'slice index starts at 102 but ends at 101', ../src/libcore/slice.rs:534` on jsgf/rocksdb-sys@bfb1a1d124d965784063dd9cea8f8d0a7bcc14d2.

The backtrace is:
```
$ RUST_BACKTRACE=1 cargo package 
   Packaging rocksdb-sys v0.2.2 (file:///home/jeremy/src/rust/rocksdb-sys)
thread '<main>' panicked at 'slice index starts at 102 but ends at 101', ../src/libcore/slice.rs:534
stack backtrace:
   1:     0x5584de3c6710 - std::sys::backtrace::tracing::imp::write::h63529ecac330b5bc
   2:     0x5584de3cf6eb - std::panicking::default_hook::_$u7b$$u7b$closure$u7d$$u7d$::h0e99487804ecdaa3
   3:     0x5584de3cf36e - std::panicking::default_hook::h1f0a2403a716c4a6
   4:     0x5584de3b3eef - std::sys_common::unwind::begin_unwind_inner::heae36551373e36bd
   5:     0x5584de3b58a8 - std::sys_common::unwind::begin_unwind_fmt::hfdca93605af5caca
   6:     0x5584de3c5941 - rust_begin_unwind
   7:     0x5584de405ddf - core::panicking::panic_fmt::h139963e0c814ece1
   8:     0x5584de405fd3 - core::slice::slice_index_order_fail::hfd97ac60d33c912a
   9:     0x5584de164fd7 - tar::header::UstarHeader::_set_path::h17b0206550960dd5
  10:     0x5584de163d66 - tar::header::Header::_set_path::h31d689d2714e1449
  11:     0x5584ddf5d374 - cargo::ops::cargo_package::package::h08cdd994f43d4c04
  12:     0x5584dde57a5f - cargo::call_main_without_stdin::h31e15b3df12e109b
  13:     0x5584dde1cc56 - cargo::execute::hab577e023b3ac5e2
  14:     0x5584dde17943 - cargo::call_main_without_stdin::hf709651485746f28
  15:     0x5584dde14aa1 - cargo::main::h2b219a79f378c1ef
  16:     0x5584de3cefe4 - std::sys_common::unwind::try::try_fn::h512e768250918f40
  17:     0x5584de3c58cb - __rust_try
  18:     0x5584de3cea2b - std::rt::lang_start::h2dd73040e4b52b40
  19:     0x7f172c54457f - __libc_start_main
  20:     0x5584dde14428 - <unknown>
```

target/package/.rocksdb-sys-0.2.2.crate exists, but it looks like its missing the following files:
```
@@ -773,29 +778,3 @@
 rocksdb/utilities/spatialdb/utils.h
 rocksdb/utilities/table_properties_collectors/compact_on_deletion_collector.cc
 rocksdb/utilities/table_properties_collectors/compact_on_deletion_collector.h
-rocksdb/utilities/table_properties_collectors/compact_on_deletion_collector_test.cc
-rocksdb/utilities/transactions/optimistic_transaction_db_impl.cc
-rocksdb/utilities/transactions/optimistic_transaction_db_impl.h
-rocksdb/utilities/transactions/optimistic_transaction_impl.cc
-rocksdb/utilities/transactions/optimistic_transaction_impl.h
-rocksdb/utilities/transactions/optimistic_transaction_test.cc
-rocksdb/utilities/transactions/transaction_base.cc
-rocksdb/utilities/transactions/transaction_base.h
-rocksdb/utilities/transactions/transaction_db_impl.cc
-rocksdb/utilities/transactions/transaction_db_impl.h
-rocksdb/utilities/transactions/transaction_db_mutex_impl.cc
-rocksdb/utilities/transactions/transaction_db_mutex_impl.h
-rocksdb/utilities/transactions/transaction_impl.cc
-rocksdb/utilities/transactions/transaction_impl.h
-rocksdb/utilities/transactions/transaction_lock_mgr.cc
-rocksdb/utilities/transactions/transaction_lock_mgr.h
-rocksdb/utilities/transactions/transaction_test.cc
-rocksdb/utilities/transactions/transaction_util.cc
-rocksdb/utilities/transactions/transaction_util.h
-rocksdb/utilities/ttl/db_ttl_impl.cc
-rocksdb/utilities/ttl/db_ttl_impl.h
-rocksdb/utilities/ttl/ttl_test.cc
-rocksdb/utilities/write_batch_with_index/write_batch_with_index.cc
-rocksdb/utilities/write_batch_with_index/write_batch_with_index_internal.cc
-rocksdb/utilities/write_batch_with_index/write_batch_with_index_internal.h
-rocksdb/utilities/write_batch_with_index/write_batch_with_index_test.cc
```

Perhaps `rocksdb/utilities/table_properties_collectors/compact_on_deletion_collector_test.cc` is too long?

**Comment by jsgf** [comment]
BTW, this is cargo installed by rustup on Linux:
```
$ cargo version
cargo 0.10.0-nightly (10ddd7d 2016-04-08)
```

**Comment by alexcrichton** [comment]
This I believed was fixed in https://github.com/alexcrichton/tar-rs/commit/d08897ae6d7a1d7a5bb381e24c5402a8d15eca46 which was published in 0.4.5 of tar-rs and brought to Cargo in 11a0a316 which was published just a few days after the version you're using. I think this is fixed on master in this case so I'm gonna close, but thanks for the report!
